udata/core/spatial/forms.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import json
+import geojson
+
+from udata.forms import widgets, ModelForm, validators
+from udata.forms.fields import ModelList, StringField, SelectField, FormField
+from udata.i18n import lazy_gettext as _
+
+from .models import GeoZone, SpatialCoverage, spatial_granularities
+
+
+class ZonesAutocompleter(widgets.TextInput):
+    classes = 'zone-completer'
+
+    def __call__(self, field, **kwargs):
+        '''Store the values as JSON to prefeed selectize'''
+        if field.data:
+            kwargs['data-values'] = json.dumps([{
+                'id': zone.id,
+                'name': zone.name
+            } for zone in field.data])
+        return super(ZonesAutocompleter, self).__call__(field, **kwargs)
+
+
+class ZonesField(ModelList, StringField):
+    model = GeoZone
+    widget = ZonesAutocompleter()
+
+
+class GeomField(StringField):
+    def process_formdata(self, valuelist):
+        if valuelist:
+            value = valuelist[0]
+            try:
+                if isinstance(value, basestring):
+                    self.data = geojson.loads(value)
+                else:
+                    self.data = geojson.GeoJSON.to_instance(value)
+            except:
```

Comment:
I can but:
- geojson raise different exceptions that do not have a common ancestor
- it will raise a 500 instead of failing the data parsing with a validation error